src/cartesian/CartesianGrid.tsx

```diff
@@ -325,10 +327,12 @@ export class CartesianGrid extends PureComponent<Props> {
 
       verticalPoints = verticalCoordinatesGenerator(
         {
-          xAxis: {
-            ...xAxis,
-            ticks: isVerticalValues ? verticalValues : xAxis.ticks,
-          },
+          xAxis: xAxis
+            ? {
+                ...xAxis,
+                ticks: isVerticalValues ? verticalValues : xAxis.ticks,
+              }
```

Comment:
Interesting, could we have caught this with better types?

Reply:
I've learned that you can't really trust our types in terms of undefined/null. In this case we should always have a y axis but because of how the components are composed you can't guarantee the user provided one. 

One way would be to always have a default axis used for calculation but not show it